### DISCUSSION TITLE
q-uploader - How to access to the file (no v-model nor value?)

### DISCUSSION THREAD

**Discussion by DocBouCou** [quasarframework/quasar#13739] [General - Components / Directives / etc] [answered]
This question is related to #13722 .

Is there a way to access to the file array hidden underneath q-uploader ? 
For q-file, we have v-model/value, it seems there is no such thing for q-uploader... So, if we need to, how can we access and manipulate the file array in q-uploader? If we want to do so, do we have to use q-file instead?

Thanks if you answer :-)

**Comment by pratik227** [comment, marked as answer]
You can use added event.

`@added -> function(files)`
Description
Emitted when files are added into the list

https://codepen.io/Pratik__007/pen/poamvWR?editors=101

https://stackoverflow.com/a/54673458

**Comment by DocBouCou** [comment]
Thank you.
So it looks like I have to bypass the q-uploader to save my file in data()...

**Reply by pratik227** [reply]
means?

**Reply by DocBouCou** [reply]
@pratik227 
In the end, what I want to do is described here: #13722 (if image, resize and compress it browser-side before upload).

So, with `@added` I have access to the file, but if I change it, it will not be saved directly in q-uploader, am I wrong? 
Thus, I should take the file using `@added`, do my logic, resize and so on, `removeFile (file)` and then `addFiles (files)` (and probably, save it in data() at some point).
Or am I missing something?

**Comment by DocBouCou** [comment]
If someone reads this discussion, just notice that the `@added` event is emitted when a file is added programmatically using `addFiles (files)`. 
So, if you want to use `@added -> removeFile -> addFiles` logic in order to modify a file (like resize an image) you have to use a flag to ignore the `@added` event when you use  `addFiles (files)`.
(@pratik227 )
